server/lib/pbench/__init__.py

```diff
@@ -216,6 +224,18 @@ def __init__(self, cfg_name):
             self.COMMIT_ID = self.conf.get("pbench-server", "commit_id")
         except NoOptionError:
             self.COMMIT_ID = "(unknown)"
+        
+        try:
+            self.logger_type = self.conf.get("logging", "logger")
+        except (NoOptionError, NoSectionError):
+            self.logger_type = "devlog"
+
+        if self.logger_type == "hostport":
```

Comment:
Consider moving this under an `else:` clause of the `try:` block of line 228, something like:

```
try:
    self.logger_type = self.conf.get("logging", "logger")
except (NoOptionError, NoSectionError):
    self.logger_type = "devlog"
else:
    if self.logger_type == "hostport":
        try:
            self.logger_host = self.conf.get("logging", "logger_host")
            self.logger_port = self.conf.get("logging", "logger_port")
        except (NoOptionError) as exc:
            # Note we don't have to check for a `NoSectionError` here because we have already successfully fetched the `logger` from the `logging` section above.
            raise BadConfig(exc)
```

Reply:
Ok sure